openlibrary/plugins/openlibrary/libraries.py

```diff
@@ -1,542 +0,0 @@
-
-    def _get_average_duration_per_month(self):
-        """Returns (date, duration-in-days) for each day with duration averaged per month."""
-        for month, chunk in itertools.groupby(self._get_average_duration_per_day(), lambda x: x[0][:2]):
-            chunk = list(chunk)
-            avg = sum(v for k, v in chunk) / len(chunk)
-            for k, v in chunk:
-                yield k, avg
-
-    def get_average_duration_per_month(self):
-        """Returns (date-in-millis, duration-in-days) for each day with duration averaged per month."""
-        return [[self.date2millis(*key), value] for key, value in self._get_average_duration_per_month()]
-
-    def get_popular_books(self, limit=10):
-        rows = self.view("loans/books", reduce=False, startkey=["", {}], endkey=[""], descending=True, limit=limit).rows
-        counts = [row.key[-1] for row in rows]
-        keys = [row.id for row in rows]
-        books = web.ctx.site.get_many(keys)
-        return zip(books, counts)
-
-    def get_loans_per_book(self, key="", limit=1):
-        """Returns the distribution of #loans/book."""
-        rows = self.view("loans/books", group=True, startkey=[key], endkey=[key, {}]).rows
-        return [[row.key[-1], row.value] for row in rows if row.value >= limit]
-
-    def get_loans_per_user(self, key="", limit=1):
-        """Returns the distribution of #loans/user."""
-        rows = self.view("loans/people", group=True, startkey=[key], endkey=[key, {}]).rows
-        return [[row.key[-1], row.value] for row in rows if row.value >= limit]
-
-    def get_loans_per_library(self):
-        counts = self._get_lib_counts()
-        return [((lib.key, lib.name), count) for lib, count in counts]
-
-    def get_loans_per_state(self):
-        counts = self._get_lib_counts()
-        return [((lib.key, lib.name), count) for lib, count in counts if lib.lending_region]
-
-    def _get_lib_counts(self):
-        # view contains:
-        #   [lib_key, status], 1
-        # Need to use group_level=1 and take key[0] to get the library key.
-        rows = self.view("loans/libraries", group=True, group_level=1).rows
-        libraries = self._get_libraries()
-        return [(libraries[row.key[0]], row.value) for row in rows if row.key[0] in libraries]
-
-    def get_active_loans_of_libraries(self):
-        """Returns count of current active loans per library as a dictionary.
-        """
-        rows = self.view("loans/libraries", group=True).rows
-        return dict((row.key[0], row.value) for row in rows if row.key[1] == "active")
-
-    def _get_library_names(self):
-        return dict((lib.key, lib.name) for lib in inlibrary.get_libraries())
-
-    def _get_libraries(self):
-        return dict((lib.key, lib) for lib in inlibrary.get_libraries())
-
-@public
-def get_active_loans_of_libraries():
-    return LoanStats().get_active_loans_of_libraries()
-
-
-def on_loan_created_statsdb(loan):
```

Comment:
:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(:(v

Reply:
Context: Looks like these methods _were_ being used to populate the Borrows graph on the homepage. The `setup` method of this file hooks to to events and calls these functions. Today we learned. Mek's got a fix in progress.

Reply:
Thank you for a more meaningful message. The relevant issue is #3194